src/Microsoft.DotNet.Build.Tasks/PackageFiles/FrameworkTargeting.targets

```diff
@@ -105,6 +105,10 @@
   <Target Name="ConvertCommonMetadataToAdditionalProperties" BeforeTargets="AssignProjectConfiguration">
     <!-- list each append as a seperate item to force re-evaluation of AdditionalProperties metadata -->
     <ItemGroup>
+      <ProjectReference>
```

Comment:
This is happening because configuration gets the OSGroup and/our TargetGroup embedded in it when the SLN uses the meta-configuration and that will leak into project references.  We need to make sure the only thing project references see is the normal configuration string (Debug/Release) and any other relevant properties will be passed through their specific Property names (OSGroup/TargetGroup) rather than leaking in through Configuration.

As such the fix is to either redefine Configuration to the simple string or not pass Configuration and pass the simple string as ConfigurationGroup.